File: mycollab-services-pro/src/main/java/com/mycollab/pro/license/LicenseInfo.java
package com.mycollab.pro.license;

import org.joda.time.LocalDateTime;

import java.util.Date;

/**
 * @author MyCollab Ltd
 * @since 5.2.6
 */
public class LicenseInfo {

    private String customerId;

    private Date issueDate;

    private Date expireDate;

    private String licenseOrg;

    private LicenseType licenseType = LicenseType.PRO_TRIAL;

    private Integer maxUsers = Integer.MAX_VALUE;

    public String getCustomerId() {
        return customerId;
    }

    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }

    public Date getIssueDate() {
        return issueDate;
    }

    public void setIssueDate(Date issueDate) {
        this.issueDate = issueDate;
    }

    public Date getExpireDate() {
        return expireDate;
    }

    public void setExpireDate(Date expireDate) {
        this.expireDate = expireDate;
    }

    public String getLicenseOrg() {
        return licenseOrg;
    }

    public void setLicenseOrg(String licenseOrg) {
        this.licenseOrg = licenseOrg;
    }

    public Integer getMaxUsers() {
        return maxUsers;
    }

    public void setMaxUsers(Integer maxUsers) {
        this.maxUsers = maxUsers;
    }

    public LicenseType getLicenseType() {
        return licenseType;
    }

    public void setLicenseType(LicenseType licenseType) {
        this.licenseType = licenseType;
    }

    public boolean isExpired() {
        return expireDate.before(new LocalDateTime().toDate());
    }

    public boolean isTrial() {
        return LicenseType.PRO_TRIAL == licenseType;
    }

    public boolean isInvalid() {
        return LicenseType.INVALID == licenseType;
    }

    public boolean isRequiredALicense() {
        return isInvalid() || (isExpired() && isTrial());
    }
}

File: mycollab-license/src/main/java/com/mycollab/license/LicenseInfo.kt
package com.mycollab.license

import java.time.LocalDate
import java.time.LocalDateTime
import java.util.*

/**
 * @author MyCollab Ltd
 * @since 5.2.6
 */
class LicenseInfo {

    var customerId: String? = null

    var issueDate: LocalDate? = null

    var expireDate: LocalDate? = null

    var licenseOrg: String? = null

    var licenseType = LicenseType.PRO_TRIAL

    var maxUsers: Int = Integer.MAX_VALUE

    val isExpired: Boolean
        get() = expireDate!!.isBefore(LocalDate.now())

    val isTrial: Boolean
        get() = LicenseType.PRO_TRIAL == licenseType

    val isInvalid: Boolean
        get() = LicenseType.INVALID == licenseType

    val isRequiredALicense: Boolean
        get() = isInvalid || isExpired && isTrial
}
